lib/generic/modules/metabolic_syndrome_disease.json

```diff
@@ -0,0 +1,651 @@
+{
+  "name" : "Metabolic Syndrome Disease Progression",
+  "remarks" : [],
+  "states" : {
+
+    "Initial" : {
+      "type" : "Initial",
+      "direct_transition" : "Initial_Kidney_Health"
+    },
+
+    "Initial_Kidney_Health" : {
+      "type" : "SetAttribute",
+      "attribute" : "diabetic_kidney_damage",
+      "value" : 0,
+      "direct_transition" : "Initial_Eye_Health"
+    },
+
+    "Initial_Eye_Health" : {
+      "type" : "SetAttribute",
+      "attribute" : "diabetic_eye_damage",
+      "value" : 0,
+      "direct_transition" : "Initial_Nerve_Health"
+    },
+
+    "Initial_Nerve_Health" : {
+      "type" : "SetAttribute",
+      "attribute" : "diabetic_nerve_damage",
+      "value" : 0,
+      "direct_transition" : "Age_and_BG_Guard"
+    },
+
+    "Age_and_BG_Guard" : {
+      "type" : "Guard",
+      "allow" : {
+      "condition_type": "And",
+      "conditions": [
+        {
+          "condition_type": "Age",
+          "operator": ">=",
+          "quantity": 16,
+          "unit": "years"
+        },
+        {
+          "condition_type" : "Vital Sign",
+          "vital_sign" : "blood_glucose",
+          "operator" : "is not nil"
+        }
+      ]
+      },
+      "direct_transition" : "Potential_Diabetes"
+    },
+
+    "Potential_Diabetes" : {
+      "type" : "Delay",
+      "exact" : { "quantity" : 7, "unit" : "days" },
+      "conditional_transition" : [
+        {
+          "condition" : {
+            "condition_type" : "And",
+            "conditions" : [
+              {
+                "condition_type" : "Attribute",
+                "attribute" : "hypertension",
+                "operator" : "is nil"
+              },
+              {
+                "condition_type": "Age",
+                "operator": ">=",
+                "quantity": 18,
+                "unit": "years"
+              }
+            ]
+          },
+          "transition" : "Chance_to_Onset_Hypertension"
+        },
+        {
+          "condition" : {
+            "condition_type" : "Vital Sign",
+            "vital_sign" : "blood_glucose",
+            "operator" : ">",
+            "value" : 6.5
+          },
+          "transition" : "Onset_Diabetes"
+        },
+        {
+          "condition" : {
+            "condition_type" : "Vital Sign",
+            "vital_sign" : "blood_glucose",
+            "operator" : ">",
+            "value" : 5.7
+          },
+          "transition" : "Onset_Prediabetes"
+        },
+        {
+          "transition" : "Potential_Diabetes"
+        }
+      ]
+    },
+
+    "Chance_to_Onset_Hypertension" : {
+      "type" : "Simple",
+      "distributed_transition" : [
+        { "distribution" : 0.594, "transition" : "Onset_Hypertension" },
```

Comment:
That state should only be hit once, when the patient is 18 and the 'hypertension' attribute is nil. It sets the flag to true or false, and then the nil check checks explicitly for nil not just a false-y value so it's not hit again